sphinxgallery/gen_rst.py

```diff
@@ -310,81 +339,57 @@ def execute_script(image_dir, thumb_file, image_fname, base_image_name,
     if os.path.exists(time_path):
         time_elapsed = float(open(time_path).read())
 
-    if not os.path.exists(first_image_file) or \
-       os.stat(first_image_file).st_mtime <= os.stat(src_file).st_mtime:
-        # We need to execute the code
-        print('plotting %s' % fname)
-        t0 = time()
-        import matplotlib.pyplot as plt
```

Comment:
@lesteve 
I'm actually moving it out of here. I asked in scikit-learn if there was a good reason to keep this import so nested in the function and not moving it to the top of the file, no answer yet.